pkg/apis/openstackproviderconfig/v1alpha1/types.go

```diff
@@ -168,6 +174,33 @@ type SubnetFilter struct {
 	NotTagsAny      string `json:"not-tags-any,omitempty"`
 }
 
+// SchedulerHints represents a set of scheduling hints that are passed to the
+// OpenStack scheduler.
+type SchedulerHints struct {
+	// Group specifies a Server Group to place the instance in.
+	Group string `json:"group,omitempty"`
+
+	// DifferentHost will place the instance on a compute node that does not
+	// host the given instances.
+	DifferentHost []string `json:"differentHost,omitempty"`
+
+	// SameHost will place the instance on a compute node that hosts the given
+	// instances.
+	SameHost []string `json:"sameHost,omitempty"`
+
+	// Query is a conditional statement that results in compute nodes able to
+	// host the instance.
+	// Store the json format of your query in this string.
+	// e.g. [">=","$free_ram_mb",1024]
+	Query string `json:"query,omitempty"`
+
+	// TargetCell specifies a cell name where the instance will be placed.
+	TargetCell string `json:"target_cell,omitempty"`
+
```

Comment:
there's definition https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/extensions/schedulerhints/requests.go#L12
we can't reuse them?

Reply:
No we can't because of `Query` and `AdditionalProperties` using `interface{}`. Which is not applicable to generate openapi specs out of.

But I forgot to rename this field to `targetCell`.